script/install/debootstrap.bash

```diff
@@ -173,5 +173,47 @@ echo "$chroot_cmd update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9
 chroot "$ISOLATE_ROOT" update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 75
 # gcc 9 done
 
-# let user know that chroot installs are finished
+[ -z "$TRAVIS" ] && bash script/confirm.bash 'Install .Net Core (C#)' && {
+  set -x
+
+  : Installing .Net Core for C#
+  # https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-ubuntu-1604
+
+  : Downloading package info from Microsoft
+  UBUNTU_VERSION=$(lsb_release --release --short)
+  chroot "$ISOLATE_ROOT" wget -q "https://packages.microsoft.com/config/ubuntu/${UBUNTU_VERSION}/packages-microsoft-prod.deb" -O packages-microsoft-prod.deb
+  chroot "$ISOLATE_ROOT" dpkg -i packages-microsoft-prod.deb
+  chroot "$ISOLATE_ROOT" rm packages-microsoft-prod.deb
+  [ "$UBUNTU_VERSION" == "18.04" ] && chroot "$ISOLATE_ROOT" sudo add-apt-repository universe
+  chroot "$ISOLATE_ROOT" apt-get install -y apt-transport-https
+  chroot "$ISOLATE_ROOT" apt-get update
+
+  : Installing dotnet 3.1
+  chroot "$ISOLATE_ROOT" apt-get install -y dotnet-sdk-3.1
+
+  : Creating runtime config file
+  # The dotnet command will not execute the compiled .exe unless there is a config file
+  # which tells it what runtime framework to use.
+  # https://stackoverflow.com/questions/46065777/is-it-possible-to-compile-a-single-c-sharp-code-file-with-the-net-core-roslyn-c
+
+  RUNTIME_CONFIG_PATH="$ISOLATE_ROOT/usr/share/dotnet-config.json"  # used in the compile command
+  RUNTIME_CONFIG_TEMPL="script/csharp/dotnet-config-template.json"
 
+  # referenced in RUNTIME_CONFIG_TEMPL file
+  DOTNET_FW_VERSION=$(dotnet --version | sed -r "s/([0-9].[0-9].[0-9])[0-9]*/\1/") \
+  DOTNET_FW_NAME="Microsoft.NETCore.App" \
+  envsubst < $RUNTIME_CONFIG_TEMPL | cat > $RUNTIME_CONFIG_PATH
+
+  : Moving shell scripts into place
+  
+  cp "script/csharp/compile-command.sh" "$ISOLATE_ROOT/usr/local/csc.sh"
+  chroot "$ISOLATE_ROOT" ln "/usr/local/csc.sh" "/usr/bin/csc"
+  chroot "$ISOLATE_ROOT" chmod +x "/usr/bin/csc"
```

Comment:
We can mark it as executable in version control, then we won't need chmod because cp will preserve the execute permission. It also makes it slightly easier to run the script directly from the version control directory.
(This might not work well on Windows, which doesn't have the notion of execute permission, but I don't think we are trying to support Windows.)

Reply:
Cool, I didn't know that